ldms/src/sampler/syspapi/syspapi_sampler.c

```diff
@@ -89,7 +89,7 @@ typedef struct syspapi_metric_s {
 	int init_rc; /* if 0, attr is good for perf_event_open() */
 	struct perf_event_attr attr; /* perf attribute */
 	char papi_name[256]; /* metric name in PAPI */
-	char pfm_name[256]; /* metric name in perfmon (PAPI native) */
+	char pfm_name[PAPI_HUGE_STR_LEN]; /* metric name in perfmon (PAPI native) */
```

Comment:
@baallan, I don't believe that this field is used anywhere. We should just remove it instead of consuming 1k for every papi metric.

Reply:
@tom95858 it is copied at line 251 into a buffer which is just 256 bytes while the papi source string is up to 1024. The alternative is to make the structure's pfm_name field separately allocated and freed, which would be  a much bigger change.

Reply:
@baallan, after it is copied in there it is never used again. My suggestion is to just remove the field instead of wasting 1k for every event.